File: ruby/classes/inheritance/base_and_derived_class.rb
#!/usr/bin/ruby2.7

class Mammal
  @classTaxon

  def initialize()
    @classTaxon = "Mammalia"
  end
end

class Fox < Mammal
  @speciesTaxon
  attr_accessor :name

  def initialize()
    @speciesTaxon = "Vulpes vulpes"
    super() # calls initialize() of base class
  end

  def show()
    print("Hi, my name is ", @name, ".\n",
      "Class: ", @classTaxon, "\n",
      "Species: ", @speciesTaxon, "\n"
    )
  end
end

foxFerdinand = Fox.new

foxFerdinand.name = "Ferdinand"
foxFerdinand.show()

File: java/classes/inheritance/BaseAndDerivedClass.java
class Mammal {
  public String classTaxon = "Mammalia";
}

class Fox extends Mammal {
  public String speciesTaxon = "Vulpes vulpes";
  public String name;

  public void show() {
    System.out.println("Hi, my name is " + name + ".\n"
      + "Class: " + classTaxon + "\n"
      + "Species: " + speciesTaxon
    );
  };
}

class BaseAndDerivedClass {
  public static void main(String[] args) {
    Fox foxFerdinand = new Fox();

    foxFerdinand.name = "Ferdinand";
    foxFerdinand.show();
  }
}
